app/src/main/java/com/tungnk123/orpheus/media/FileSystemSongProvider.kt

```
@@ -0,0 +1,185 @@
+    private fun getMediaFolders(): List<Uri> {
+        val folders = listOf(Uri.parse("file:///storage/emulated/0/Music"))
+        Log.d(TAG, "Media folders: $folders")
+        return folders
+    }
+
+    @RequiresApi(Build.VERSION_CODES.S)
+    private suspend fun scanMediaTree(path: SimplePath, file: DocumentFileX) {
+        try {
+            Log.d(TAG, "Scanning media tree: $path")
+            coroutineScope {
+                file.list().map {
```

detekt (1.23.7)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)